### PR TITLE
llmq: Store encrypted DKG contributitons 

### PR DESCRIPTION
**Pull request by xdustinface**
This is like #3937 (where this one is based on) also a preparation for the next coming PR which will use this encrypted contributions after DKG to share them with other members.

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
Needs rebase (probably)

File: src/llmq/quorums_dkgsession.cpp [hunk 1]
```diff
@@ -383,6 +385,9 @@ void CDKGSession::VerifyPendingContributions()
         memberIndexes.emplace_back(idx);
         vvecs.emplace_back(receivedVvecs[idx]);
         skContributions.emplace_back(receivedSkContributions[idx]);
+        // Note: Write here already to definitly store one contribution for each member no matter if
```
Comment (thephez) [inline src/llmq/quorums_dkgsession.cpp:388]:
```suggestion
        // Note: Write here already to definitely store one contribution for each member no matter if
```

File: src/llmq/quorums_dkgsession.cpp [hunk 2]
```diff
@@ -383,6 +385,9 @@ void CDKGSession::VerifyPendingContributions()
         memberIndexes.emplace_back(idx);
         vvecs.emplace_back(receivedVvecs[idx]);
         skContributions.emplace_back(receivedSkContributions[idx]);
+        // Note: Write here already to definitely store one contribution for each member no matter if
+        // our share is valid or not, "could" be that others are still correct?
```
Comment (PastaPastaPasta) [inline src/llmq/quorums_dkgsession.cpp:389]: See 8db3ef614a371836289a0cd69ea97f212705342f

Just simplifies the language/wording of the comment

File: src/llmq/quorums_dkgsession.h [hunk 3]
```diff
@@ -262,6 +262,7 @@ class CDKGSession
     std::vector<BLSVerificationVectorPtr> receivedVvecs;
     // these are not necessarily verified yet. Only trust in what was written to the DB
     BLSSecretKeyVector receivedSkContributions;
+    std::vector<std::shared_ptr<CBLSIESMultiRecipientObjects<CBLSSecretKey>>> vecEncryptedContributions;
```
Comment (PastaPastaPasta) [inline src/llmq/quorums_dkgsession.h:265]: Please add a doxygen comment for this list

File: src/llmq/quorums_dkgsessionmgr.h [hunk 4]
```diff
@@ -61,10 +61,12 @@ class CDKGSessionManager
...
 
-    // Verified contributions are written while in the DKG
+    // Contributions are written while in the DKG
     void WriteVerifiedVvecContribution(Consensus::LLMQType llmqType, const CBlockIndex* pindexQuorum, const uint256& proTxHash, const BLSVerificationVectorPtr& vvec);
     void WriteVerifiedSkContribution(Consensus::LLMQType llmqType, const CBlockIndex* pindexQuorum, const uint256& proTxHash, const CBLSSecretKey& skContribution);
     bool GetVerifiedContributions(Consensus::LLMQType llmqType, const CBlockIndex* pindexQuorum, const std::vector<bool>& validMembers, std::vector<uint16_t>& memberIndexesRet, std::vector<BLSVerificationVectorPtr>& vvecsRet, BLSSecretKeyVector& skContributionsRet);
+    void WriteEncryptedContributions(Consensus::LLMQType llmqType, const CBlockIndex* pindexQuorum, const uint256& proTxHash, const CBLSIESMultiRecipientObjects<CBLSSecretKey>& contributions);
+    bool GetEncryptedContributions(Consensus::LLMQType llmqType, const CBlockIndex* pindexQuorum, const std::vector<bool>& validMembers, const uint256& proTxHash, std::vector<CBLSIESEncryptedObject<CBLSSecretKey>>& vecRet);
```
Comment (PastaPastaPasta) [inline src/llmq/quorums_dkgsessionmgr.h:69]: Please add doxygen comments for these methods

**Review by PastaPastaPasta — COMMENTED** [review]
Looks good overall

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**